src/reconstruction/reconstruct-asct-b.js

```diff
@@ -0,0 +1,335 @@
+  
+  const newHeader = Array.from(allColumns).sort((a, b) => {
+    // Get the prefix of each column (e.g., "AS/1" -> "AS", "CT/2/LABEL" -> "CT")
+    const getPrefix = (col) => {
+      const parts = col.split('/');
+      return parts[0];
+    };
+    
+    // Get the base column name (e.g., "AS/1/LABEL" -> "AS/1", "CT/2/ID" -> "CT/2")
+    const getBaseColumn = (col) => {
+      const parts = col.split('/');
+      if (parts.length >= 2) {
+        return `${parts[0]}/${parts[1]}`;
+      }
+      return col;
+    };
+    
+    // Get the suffix type (e.g., "AS/1/LABEL" -> "LABEL", "CT/2/ID" -> "ID", "AS/1" -> "")
+    const getSuffixType = (col) => {
+      const parts = col.split('/');
+      if (parts.length >= 3) {
+        return parts[2];
+      }
+      return '';
+    };
+    
+    const prefixA = getPrefix(a);
+    const prefixB = getPrefix(b);
+    
+    // Find the order of each prefix in our desired order
+    const orderA = columnOrder.indexOf(prefixA);
+    const orderB = columnOrder.indexOf(prefixB);
+    
+    // If both prefixes are in our order list, sort by their position
+    if (orderA !== -1 && orderB !== -1) {
+      if (orderA !== orderB) {
+        return orderA - orderB;
+      }
+      
+      // If same prefix, sort by base column first, then by suffix type
+      const baseColumnA = getBaseColumn(a);
+      const baseColumnB = getBaseColumn(b);
+      
+      if (baseColumnA !== baseColumnB) {
+        return baseColumnA.localeCompare(baseColumnB);
+      }
+      
+      // If same base column, sort by suffix type: base column first, then LABEL, then ID
+      const suffixA = getSuffixType(a);
+      const suffixB = getSuffixType(b);
+      
+      const suffixOrder = { '': 0, 'LABEL': 1, 'ID': 2 };
+      const suffixOrderA = suffixOrder[suffixA] ?? 3;
+      const suffixOrderB = suffixOrder[suffixB] ?? 3;
+      
+      return suffixOrderA - suffixOrderB;
+    }
+    
+    // If only one is in our order list, prioritize it
+    if (orderA !== -1 && orderB === -1) return -1;
+    if (orderA === -1 && orderB !== -1) return 1;
+    
+    // If neither is in our order list, sort alphabetically
+    return a.localeCompare(b);
+  });
+
+  // Create metadata rows
+  const metadataRows = [
+    [metadata['?tableTitle'], '', '', '', '', '', '', '', ''], // First row with table title
+    ['', '', '', '', '', '', '', '', ''], // Empty second row
+    [`Author Name(s):`, metadata['?authorNames']],
+    [`Author ORCID(s):`, metadata['?authorOrcids']],
+    [`Reviewer(s):`, metadata['?reviewerNames']],
+    [`Reviewer ORCID(s):`, metadata['?reviewerOrcids']],
+    [`General Publication(s):`, metadata['?generalPublications']],
+    [`Data DOI:`, metadata['?dataDoi']],
+    [`Date:`, metadata['?date']],
+    [`Version Number:`, metadata['?versionNumber']]
+  ];
+
+  // Write output with metadata at the top
+  const csvDataRows = transformedRows.map(row => newHeader.map(col => `"${row[col] || ''}"`).join(','));
```

Comment:
Don't try to write your own csv writer. Please use papaparse as is used elsewhere. Doing this by hand is a recipe for disaster.